src/mame/audio/nl_gunfight.cpp

```diff
@@ -0,0 +1,492 @@
+
+#endif
+
+	// The following are the current inputs to the storage capacitors and
+	// sound-effect amplifiers.
+
+	RES(R130, RES_R(15))
+	RES(R230, RES_R(15))
+	RES(R117, RES_R(15))
+	RES(R217, RES_R(15))
+
+	CAP(C122, CAP_U(10))
+	CAP(C222, CAP_U(10))
+	CAP(C116, CAP_U(20))
+	CAP(C216, CAP_U(20))
+	NET_C(C122.2, C222.2, C116.2, C216.2, GND)
+
+	NET_C(R130.2, C122.1, R126.1, R124.1)
+	NET_C(R230.2, C222.1, R226.1, R224.1)
+	NET_C(R117.2, C116.1, R113.1, R111.1)
+	NET_C(R217.2, C216.1, R213.1, R211.1)
+
+	// Use a simple noise generating macro device, based on a voltage
+	// source, to simulate the AC noise generated by amplified Zener.
+	// This is the basic noise source which is filtered and amplified by
+	// the sound-effect circuits.
+
+	NOISE(N1)
+	NET_C(N1.N, GND)
+
+	// Coupling capacitors from noise generator to sound effect
+	// frequency filters.
+
+	CAP(C303, CAP_U(0.1))
+	CAP(C306, CAP_U(0.1))
+	CAP(C304, CAP_U(0.1))
+	CAP(C305, CAP_U(0.1))
+
+	NET_C(C303.1, C306.1, C304.1, C305.1, N1.P)
+
+	// Sound effect frequency filters. Each of these is a second-order
+	// low-pass filter with cutoff frequency determined by its component
+	// values. The different capacitor values produce each sound effect's
+	// distinct pitch.
+
+	RES(R129, RES_K(20))
+	RES(R229, RES_K(20))
+	RES(R116, RES_K(20))
+	RES(R216, RES_K(20))
+
+	NET_C(C303.2, R129.1)
+	NET_C(C306.2, R229.1)
+	NET_C(C304.2, R116.1)
+	NET_C(C305.2, R216.1)
+
+	CAP(C121, CAP_U(0.015))
+	CAP(C221, CAP_U(0.015))
+	CAP(C115, CAP_U(0.033))
+	CAP(C215, CAP_U(0.033))
+	NET_C(C121.2, C221.2, C115.2, C215.2, GND)
+
+	RES(R128, RES_K(30))
+	RES(R228, RES_K(30))
+	RES(R115, RES_K(30))
+	RES(R215, RES_K(30))
+
+	NET_C(R129.2, C121.1, R128.1)
+	NET_C(R229.2, C221.1, R228.1)
+	NET_C(R116.2, C115.1, R115.1)
+	NET_C(R216.2, C215.1, R215.1)
+
+	CAP(C120, CAP_U(0.01))
+	CAP(C220, CAP_U(0.015))
+	CAP(C114, CAP_U(0.1))
+	CAP(C214, CAP_U(0.047))
+	NET_C(C120.2, C220.2, C114.2, C214.2, GND)
+
+	// Sound effect amplifier circuits. Each of these is a single NPN
+	// transistor wired as a common-emitter amplifier. The "hit"
+	// sound effect amplifiers also have a bypass capacitor at the
+	// emitter. The attack and decay of the sound effects is handled
+	// by controlling the current supply to each amplifier, which is
+	// done by the switching circuits described above.
+
+	RES(R126, RES_K(330))
+	RES(R226, RES_K(330))
+	RES(R113, RES_K(330))
+	RES(R213, RES_K(330))
+
+	RES(R127, RES_K(30))
+	RES(R227, RES_K(30))
+	RES(R114, RES_K(30))
+	RES(R214, RES_K(30))
+	NET_C(R127.2, R227.2, R114.2, R214.2, GND)
+
+	RES(R124, RES_K(5.1))
+	RES(R224, RES_K(5.1))
+	RES(R111, RES_K(5.1))
+	RES(R211, RES_K(5.1))
+
+	RES(R125, RES_R(100))
+	RES(R225, RES_R(100))
+	RES(R112, RES_K(1))
+	RES(R212, RES_K(1))
+	NET_C(R125.2, R225.2, R112.2, R212.2, GND)
+
+	CAP(C113, CAP_U(50))
+	CAP(C213, CAP_U(50))
+	NET_C(C113.2, C213.2, GND)
+
+	QBJT_EB(Q106, "NPN")
+	QBJT_EB(Q206, "NPN")
+	QBJT_EB(Q103, "NPN")
+	QBJT_EB(Q203, "NPN")
```

Comment:
Is there any reason you’re not using a specific transistor model?  The netlist library accepts SPICE transistor models, and they’re fairly easy to add.

Reply:
I'd considered that, but the actual transistors in question, the Amperex A-138, are actually rather obscure, as I mention in a comment block near the top of that file. I decided it would be safe to go generic for now; a specific model can always be added later. I would think that a circuit like this should work fine with a generic transistor model as long as that model is acceptable for low-power audio-frequency work. Because the transistors used in these circuits were themselves quite variable in properties from one component to the next, the amplifier circuits were designed to be independent of such variations.

Reply:
In retrospect, I should have paid closer attention to this question, since the specifics of the transistor had much more impact on the shot sounds than I originally thought—see PR #6958. Just wanted to point out that cuavas made a very good call here.